gc/base/Configuration.cpp

```diff
@@ -511,16 +511,22 @@ MM_Configuration::createParallelDispatcher(MM_EnvironmentBase *env, omrsig_handl
 
 #if defined(J9VM_OPT_CRIU_SUPPORT)
 void
-MM_Configuration::adjustGCThreadCountOnCheckpoint(MM_EnvironmentBase* env)
+MM_Configuration::adjustGCThreadCountForCheckpoint(MM_EnvironmentBase* env)
 {
 	MM_GCExtensionsBase* extensions = env->getExtensions();
+	MM_ParallelDispatcher* dispatcher = extensions->dispatcher;
 
-	extensions->dispatcher->contractThreadPool(env, extensions->checkpointGCthreadCount);
+	dispatcher->contractThreadPool(env, extensions->checkpointGCthreadCount);
+	_delegate.setMaxGCThreadCount(env, dispatcher->getPoolMaxCapacity());
```

Comment:
add a comment, that this line is temporary, to prevent growth thread pool beyond the one at snapshot side till more dependent functional components (beyond just Dispatcher) are ensured to be ok with the growth